enterprise/server/scheduling/task_router/task_router.go

```diff
@@ -0,0 +1,187 @@
+package task_router
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/platform"
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
+	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
+	"github.com/buildbuddy-io/buildbuddy/server/util/perms"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"github.com/go-redis/redis/v8"
+	"github.com/golang/protobuf/proto"
+
+	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+)
+
+const (
+	day = 24 * time.Hour
+
+	// The TTL for each node list in the routing table.
+	routingPropsKeyTTL = 7 * day
+
+	// The default max number of preferred nodes that are returned by the task
+	// router for routable tasks. This is intentionally less than the number of
+	// probes per task (for load balancing purposes).
+	defaultPreferredNodeLimit = 1
+	// The preferred node limit for workflows. This should be at least as big as
+	// the number of probes per task, since we strongly prefer to route workflow
+	// actions to executors that ran them previously.
+	workflowsPreferredNodeLimit = 6
```

Comment:
why not 3?

Reply:
It's a bit of extra padding in case a node goes down or a probe fails.